geometry/include/pcl/geometry/mesh_indices.h

```diff
@@ -426,167 +166,43 @@ class EdgeIndex
   int index_;
 
   friend std::istream&
-  operator>>(std::istream& is, pcl::geometry::EdgeIndex& index);
+  operator>> <MeshIndexT>(std::istream& is, pcl::detail::MeshIndex<MeshIndexT>& index);
 };
 
 /** \brief ostream operator. */
+template<class MeshIndexT>
 inline std::ostream&
-operator<<(std::ostream& os, const pcl::geometry::EdgeIndex& index)
+operator<<(std::ostream& os, const pcl::detail::MeshIndex<MeshIndexT>& index)
 {
   return (os << index.get());
 }
 
 /** \brief istream operator. */
+template<class MeshIndexT>
 inline std::istream&
-operator>>(std::istream& is, pcl::geometry::EdgeIndex& index)
+operator>>(std::istream& is, pcl::detail::MeshIndex<MeshIndexT>& index)
 {
   return (is >> index.index_);
 }
 
-} // End namespace geometry
-} // End namespace pcl
-
-////////////////////////////////////////////////////////////////////////////////
-// FaceIndex
-////////////////////////////////////////////////////////////////////////////////
+using VertexIndex = MeshIndex<struct VertexIndexT>;
```

Comment:
PCL uses `PrefixT` in a lot of places like `PointT` to mean it's a template type argument for a point. To prevent confusion, can we rename to `VertexIndexTag` or similar?